### PR TITLE
both sides of productions added, some visualization added

### PR DESCRIPTION
**Pull request by Loloxon**

### CODE REVIEW

File: graph/hypergraph.py [hunk 1]
```diff
@@ -129,8 +129,10 @@ def parse_networkx_to_hypergraph(nx_graph: nx.Graph) -> 'HyperGraph':
             is_border = data.get('is_border', False)
             label = data.get('label', 'E')
             # Find corresponding nodes in the nodes list
-            node1_obj = next(node for node in nodes if node.x == nx_graph.nodes[node1]['x'] and node.y == nx_graph.nodes[node1]['y'])
-            node2_obj = next(node for node in nodes if node.x == nx_graph.nodes[node2]['x'] and node.y == nx_graph.nodes[node2]['y'])
+            node1_obj = next(
+                node for node in nodes if node.x == nx_graph.nodes[node1]['x'] and node.y == nx_graph.nodes[node1]['y'])
+            node2_obj = next(
+                node for node in nodes if node.x == nx_graph.nodes[node2]['x'] and node.y == nx_graph.nodes[node2]['y'])
```
Comment (strjakub) [inline graph/hypergraph.py:135]:
```suggestion
            node1_obj = next(
                node for node in nodes if node.x == nx_graph.nodes[node1]['x'] and node.y == nx_graph.nodes[node1]['y']
            )
            node2_obj = next(
                node for node in nodes if node.x == nx_graph.nodes[node2]['x'] and node.y == nx_graph.nodes[node2]['y']
            )
```

File: graph/hypergraph.py [hunk 2]
```diff
@@ -144,53 +146,139 @@ def parse_networkx_to_hypergraph(nx_graph: nx.Graph) -> 'HyperGraph':
                 for neighbor in nx_graph.neighbors(node):
                     if not isinstance(neighbor, str):  # Regular node (not a hyperedge node)
                         neighbor_obj = next(
-                            n for n in nodes if n.x == nx_graph.nodes[neighbor]['x'] and n.y == nx_graph.nodes[neighbor]['y'])
+                            n for n in nodes if
+                            n.x == nx_graph.nodes[neighbor]['x'] and n.y == nx_graph.nodes[neighbor]['y'])
```
Comment (strjakub) [inline graph/hypergraph.py:150]:
```suggestion
                            n for n in nodes if
                                n.x == nx_graph.nodes[neighbor]['x'] and n.y == nx_graph.nodes[neighbor]['y']
                            )
```

File: graph/hypergraph.py [hunk 3]
```diff
@@ -144,53 +146,139 @@ def parse_networkx_to_hypergraph(nx_graph: nx.Graph) -> 'HyperGraph':
...
+
+        # Draw non-border edges (dotted)
+        nx.draw_networkx_edges(
+            graph,
+            pos,
+            edgelist=inside_edges,
+            style="solid",
+            width=1,
+            edge_color="grey"
+        )
```
Comment (strjakub) [inline graph/hypergraph.py:275]: comment says `dotted`, but `solid` passed inside

File: productions/p1.py [hunk 4]
```diff
@@ -1,22 +1,64 @@
...
     def _right_side(self) -> HyperGraph:
+        nodes = [Node(0, 0, False, "V"),
+                 Node(0.5, 0, True, "V"),
+                 Node(1, 0, False, "V"),
+                 Node(0, 0.5, True, "V"),
+                 Node(0.5, 0.5, False, "V"),
+                 Node(1, 0.5, True, "V"),
+                 Node(0, 1, False, "V"),
+                 Node(0.5, 1, True, "V"),
+                 Node(1, 1, False, "V")]
```
Comment (strjakub) [inline productions/p1.py:40]: just making sure, passing here explicit coordinates won't break anything later, right?

**Review by strjakub — APPROVED** [review]

File: graph/hypergraph.py [hunk 5]
```diff
@@ -144,53 +146,139 @@ def parse_networkx_to_hypergraph(nx_graph: nx.Graph) -> 'HyperGraph':
...
+        nx.draw_networkx_edges(
+            graph,
+            pos,
+            edgelist=non_border_edges,
+            style="dotted",
+            width=2,
+            edge_color="black"
+        )
+
+        # Draw non-border edges (dotted)
```
Comment (Loloxon) [inline graph/hypergraph.py:267]: `# Draw non-border edges (solid)`

Comment (Loloxon) [inline graph/hypergraph.py:267]:
```suggestion
        # Draw non-border edges (solid)
```

File: graph/hypergraph.py [hunk 6]
```diff
@@ -144,53 +146,139 @@ def parse_networkx_to_hypergraph(nx_graph: nx.Graph) -> 'HyperGraph':
...
+        nx.draw_networkx_edges(
+            graph,
+            pos,
+            edgelist=inside_edges,
```
Comment (Loloxon) [inline graph/hypergraph.py:271]:
```suggestion
        # Draw edges to hyperedge node (solid, grey)
        nx.draw_networkx_edges(
            graph,
            pos,
            edgelist=inside_edges,
```

**Pull request merged**